### PR TITLE
Add documentation about reloading search analyzers

### PR DESCRIPTION
**Pull request by cbuescher**
This PR adds documentation to the feature being worked on in #42669

### CODE REVIEW

File: docs/reference/indices/reload-analyzers.asciidoc [hunk 1]
```diff
@@ -0,0 +1,56 @@
+[[indices-reload-analyzers]]
```
Comment (jimczi) [inline docs/reference/indices/reload-analyzers.asciidoc]: This should be marked as experimental.

File: docs/reference/analysis/tokenfilters/synonym-tokenfilter.asciidoc [hunk 2]
```diff
@@ -181,3 +185,40 @@ error.
...
+        }
+    }
+}
+--------------------------------------------------
+// CONSOLE
+
+Using the <<analyzer reload API,indices-reload-analyzers>>, you can trigger reloading of the
+synonym definition. The contents of the configured synonyms file will be reloaded and the
+synonyms definition the filter uses will be updated. Note that if you trying to use the above
+analyzer as an index analyzer will result in an error.
```
Comment (jimczi) [inline docs/reference/analysis/tokenfilters/synonym-tokenfilter.asciidoc]: I would extract the last sentence in a NOTE in order to make sure that users are aware of the limitation.

File: docs/reference/indices/reload-analyzers.asciidoc [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    }
+}
+--------------------------------------------------
+// CONSOLE
+
+<1> Mark the synonym filter as updateable.
+<2> Synonym filter is usable as a search_analyzer.
+
+Calling the `_reload_search_analyzers` endpoint will now trigger reloading the
+synonyms from the configured "synonym.txt" file.
```
Comment (jimczi) [inline docs/reference/indices/reload-analyzers.asciidoc]: Maybe add a NOTE regarding the fact that the reload is per node so you need to update your `synonym.txt` file on every node **before** calling reload.

**Review by jimczi — COMMENTED** [review]
Thanks @cbuescher . I left some comments.

**Comment by cbuescher** [comment]
@jimczi I will also reopen this against the new feature branch I had to create and address your comments there.

**Pull request closed without merging**